lib/systems/include/BaseLaunchController.h

```diff
@@ -37,19 +40,33 @@ class BaseLaunchController : public virtual Controller
     int16_t init_speed_target_ = 0;
 
 public:
+    /// @brief Constructor for template launch controller
+    /// @param initial_speed_target unused right now
+    /// @note requires one method: calc_launch_algo
     BaseLaunchController(int16_t initial_speed_target)
         : init_speed_target_(initial_speed_target)
     {
         writeout_.command = BaseControllerParams::TC_COMMAND_NO_TORQUE;
         writeout_.ready = true;
     }
 
+    /// @brief ticks launch controller to progress through launch states when conditions are met
+    /// @param SysTick_s &tick 
+    /// @param PedalsSystemData_s &pedalsData 
+    /// @param float[] wheel_rpms 
+    /// @param VectornavData_s &vn_data 
     void tick(const SysTick_s &tick,
               const PedalsSystemData_s &pedalsData,
               const float wheel_rpms[],
               const VectornavData_s &vn_data);
     LaunchStates_e get_launch_state() { return launch_state_; }
+    /// @brief calculates how speed target is set among launch controllers
+    /// @param VectornavData_s &vn_data 
+    /// @note has important variation in launch controller tick as the launch controllers share a tick method
     virtual void calc_launch_algo(const VectornavData_s &vn_data) = 0;
```

Comment:
Where is this used/implemented? Calculate launch algorithm is what I'm assuming this is. What does it mean for this function to "calculate how speed target is set among launch controllers"? Do you mean that this is an abstract function that each particular launch controller must implement?

What does that @note mean? What has an "important variation" in launch controller? What does it mean for launch controllers to "share" a tick method? Is it that they all implement the same parent tick() function implementation, or simply that they all have their own implementation of the tick() function?

Reply:
Since this function's main functionality is that it updates the **state** of this launch controller (by updating a member variable, rather than returning a value), I'd really like that to be included in the comment here.